File: backend/src/main/kotlin/com/projectticket/ticket/reservation/ReservationStatus.kt
package com.projectticket.ticket.reservation

/**
 * 예매의 상태. `reservation.status` 의 값이고 `reservation_status_check` 가 같은 목록을 든다(`D14` 「열거값」).
 * 전이는 `state-machines.md`(D3)가 정하고 `reservation_status_transition` 트리거가 문지기다.
 */
enum class ReservationStatus {
    HELD, PAYING, RESERVED, CANCELLED, EXPIRED;

    val code: String get() = name.lowercase()

    companion object {
        /** 모르는 값이면 터진다. 조용히 넘기면 새 상태가 제약에 늘었을 때 그 예매만 이유 없이 화면에서 빠진다 */
        fun of(code: String): ReservationStatus =
            entries.firstOrNull { it.code == code } ?: throw IllegalStateException("모르는 예매 상태다: $code")
    }
}

File: frontend/src/lib/reservation-status.ts
/**
 * 예매 상태의 화면 이름(`44b`, `D3` 예매 상태기계). **내부 값을 그대로 안 보인다**(`D17`) — `reserved` 가 아니라 「예매 완료」.
 * 모르는 값은 그대로 두지 않고 「확인 중」으로 — 서버가 상태를 더했는데 화면이 모르면 영문 코드가 새어 나간다.
 */
const LABELS: Record<string, string> = {
  held: "좌석 선점 중",
  paying: "결제 처리 중",
  reserved: "예매 완료",
  cancelled: "취소됨",
  expired: "선점 시간 만료",
};

export function reservationStatusLabel(status: string): string {
  return LABELS[status] ?? "확인 중";
}
